build.sh

```diff
@@ -121,14 +121,32 @@ else
   $PYTHON_EXECUTABLE -m pip install \
       --target=$ROOT_DIR/python/ray/pyarrow_files pyarrow==0.12.0.RAY \
       --find-links https://s3-us-west-2.amazonaws.com/arrow-wheels/9357dc130789ee42f8181d8724bee1d5d1509060/index.html
-  bazel build //:ray_pkg -c opt
-  # Copy files and keep them writeable. This is a workaround, as Bazel
-  # marks all generated files non-writeable. If we would just copy them
-  # over without adding write permission, the copy would fail the next time.
-  # TODO(pcm): It would be great to have a solution here that does not
-  # require us to copy the files.
-  find $ROOT_DIR/bazel-genfiles/ray_pkg/ -exec chmod +w {} \;
-  cp -r $ROOT_DIR/bazel-genfiles/ray_pkg/ray $ROOT_DIR/python || true
+  
+if [ "$RAY_BUILD_JAVA" == "YES" ]; then
+    bazel build //:ray_java_pkg -c opt
```

Comment:
The arrow python package is now downloaded from AWS. It looks like arrow Java package still goes with the old way, say, downloading source files from GitHub and build locally. I remember there is a special fix for arrow to enable bazel. Is that special fix only for Python? @pcmoritz

Reply:
We download the pyarrow files from AWS because it is compile differently from the official pyarrow package. This only concerns the python part (the official package is not compatible with the official tensorflow wheels).